AMW_rest/src/main/java/ch/mobi/itc/mobiliar/rest/environments/EnvironmentsRest.java

```diff
@@ -50,4 +57,36 @@ public List<EnvironmentDTO> getContexts() {
         return contextLocator.getAllEnvironments().stream().map(EnvironmentDTO::new).collect(Collectors.toList());
     }
 
+    @POST
+    @Path("/contexts")
+    @ApiOperation(value = "Add new context")
+    public Response addContext(@ApiParam() EnvironmentDTO request) {
+        try {
+            environmentsScreenDomainService.createContextByName(request.getName(), request.getParentId());
+            return Response.status(Response.Status.OK).build();
+        } catch (Exception e) {
+            return Response.status(Response.Status.BAD_REQUEST).entity(Collections.singletonMap("message", e.getMessage())).build();
+        }
+
+    }
+
+    @PUT
+    @Path("/contexts/{id : \\d+}")
+    @ApiOperation(value = "Update existing context")
+    public Response updateContext(@ApiParam("Environment ID") @PathParam("id") Integer id, EnvironmentDTO request) {
+        try {
+            environmentsScreenDomainService.saveEnvironment(id, request.getName(), request.getNameAlias());
+            return Response.status(Response.Status.OK).build();
+        } catch (Exception e) {
```

Comment:
do not catch all exceptions here.
environmentScreenDomainService should only throw `ResourceNotFoundExceptions` 

This should already be covered by `NoResultExceptionMapper`